barretenberg/cpp/pil/vm2/context.pil

```diff
@@ -1,17 +1,79 @@
 // This is a virtual gadget, which is part of the execution trace.
 namespace execution;
 
+    // Useful to define some opcodes within this pil file
+    // Constrained to be boolean by execution instruction spec table
+    pol commit call_sel;
+    pol commit static_call_sel;
+    pol CALL = call_sel + static_call_sel; // Guaranteed to be mutually exclusive
+    // CALL & precomputed.first_row are NAND
+    CALL * precomputed.first_row = 0;
+
+    // Context columns
     pol commit context_id;
+    pol commit parent_id;
     pol commit pc;
+    pol commit next_pc;
     pol commit msg_sender;
     pol commit contract_address;
-    
-    pol commit is_static;
-    is_static * (1 - is_static) = 0;
+    pol commit is_static; // Constrained boolean by tx trace
 
     pol commit parent_calldata_offset_addr;
     pol commit parent_calldata_size_addr;
 
     pol commit last_child_returndata_offset_addr;
     pol commit last_child_returndata_size_addr;
     pol commit last_child_success; // Careful with this for now...
+
+    // ==== Helper columns ====
+    // TODO: These context modifiers also need to factor in when a new enqueued call occurs
+    // REPLACE prefixed precomputed.first_row in relations with actual phase / enqueued call change
+
+    // next_context_id increments with each invocation of an external call or new enqueued call
+    pol commit next_context_id; // Can be replaced by clk
+    // The initial next_context_id = 2, in row = 1
+    #[INCR_CONTEXT_ID]
+    (1- precomputed.first_row) * sel' * (next_context_id' - (next_context_id + CALL)) = 0;
+
+    // CALL = 1 <==> context_id' = next_context_id
+    // CALL = 0 <==> context_id' = context_id
```

Comment:
Does this hold on return?

Reply:
These two comments apply to the relation that follows, right? NEXT_CONTEXT_ID?

Reply:
yep!